src/openrct2/ride/Vehicle.cpp

```diff
@@ -2558,102 +2558,112 @@ void Vehicle::UpdateWaitingToDepart()
 
     switch (curRide->mode)
     {
-        case RIDE_MODE_DODGEMS:
+        case RideMode::Dodgems:
             // Dodgems mode uses sub_state / var_CE to tell how long
             // the vehicle has been ridden.
             SetState(Vehicle::Status::TravellingDodgems);
             var_CE = 0;
             UpdateDodgemsMode();
             break;
-        case RIDE_MODE_SWING:
+        case RideMode::Swing:
             SetState(Vehicle::Status::Swinging);
             var_CE = 0;
             current_time = -1;
             UpdateSwinging();
             break;
-        case RIDE_MODE_ROTATION:
+        case RideMode::Rotation:
             SetState(Vehicle::Status::Rotating);
             var_CE = 0;
             current_time = -1;
             UpdateRotating();
             break;
-        case RIDE_MODE_FILM_AVENGING_AVIATORS:
+        case RideMode::FilmAvengingAviators:
             SetState(Vehicle::Status::SimulatorOperating);
             current_time = -1;
             UpdateSimulatorOperating();
             break;
-        case RIDE_MODE_FILM_THRILL_RIDERS:
+        case RideMode::FilmThrillRiders:
             SetState(Vehicle::Status::SimulatorOperating, 1);
             current_time = -1;
             UpdateSimulatorOperating();
             break;
-        case RIDE_MODE_BEGINNERS:
-        case RIDE_MODE_INTENSE:
-        case RIDE_MODE_BERSERK:
+        case RideMode::Beginners:
+        case RideMode::Intense:
+        case RideMode::Berserk:
             SetState(Vehicle::Status::TopSpinOperating, sub_state);
 
             switch (curRide->mode)
             {
-                case RIDE_MODE_BEGINNERS:
+                case RideMode::Beginners:
                     sub_state = 0;
                     break;
-                case RIDE_MODE_INTENSE:
+                case RideMode::Intense:
                     sub_state = 1;
                     break;
-                case RIDE_MODE_BERSERK:
+                case RideMode::Berserk:
                     sub_state = 2;
                     break;
+                default:
+                {
+                    // This is workaround for multiple compilation errors of type "enumeration value ‘RIDE_MODE_*' not handled
+                    // in switch [-Werror=switch]"
+                }
```

Comment:
I'm a bit torn here, we don't want to handle all cases with default, nor listing all of them seem appropriate. I'm thinking we could just use ternary:
```c++
sub_state = curRide->mode == RideMode::Beginners ? 0 : curRide->mode == RideMode::Intense ? 1 : 2;
```